trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/credentials/TestSigningController.java

```diff
@@ -29,10 +28,10 @@
 
 public class TestSigningController
```

Comment:
can we add a test that uses `SigningService.STS` and some non-empty entity

Reply:
`TestProxiedAssumedRoleRequests` actually does that. It's buried in the AWS client but `TestProxiedAssumedRoleRequests` builds an `StsAssumeRoleCredentialsProvider` that uses an `StsClient` to assume a role. This was my main intent with this change. This will mesh neatly with standard user workflows. They will all know how to do this already.

So, look at `TestProxiedAssumedRoleRequests.buildClient()`. You can then run that test and put a breakpoint in `Signer.sign()` and you will see that the entity is non-empty. It's the form-encoded STS request. Something like:

```
Action=AssumeRole&Version=2011-06-15&RoleArn=arn%3Aaws%3Aiam%3A%3A123456789012%3Arole%2Fassumed&RoleSessionName=dummy&ExternalId=dummy
```

Reply:
Ah nice, thanks for clarifying!